docs/en/integrations/data-visualization/grafana/config.md

```diff
@@ -0,0 +1,234 @@
+---
+sidebar_label: Plugin Configuration
+sidebar_position: 3
+slug: /en/integrations/grafana/config
+description: Configuration options for the ClickHouse data source plugin in Grafana
+---
+import ConnectionDetails from '@site/docs/en/_snippets/_gather_your_details_native.md';
+
+# Configuring ClickHouse data source in Grafana
+
+The easiest way to modify a configuration is in the Grafana UI on the plugin configuration page, but data sources can also be [provisioned with a YAML file](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources).
+
+This page shows a list of options available for configuration in the ClickHouse plugin, as well as config snippets for those provisioning a data source with YAML.
+
+For a quick overview of all the options, a full example config can be found [here](#full-example-config).
+
+## Common Settings
+
+Example configuration screen:
+<img src={require('./images/config_common.png').default} class="image" alt="Example secure native config" />
+
+Example configuration YAML for common settings:
+```yaml
+jsonData:
+  host: 127.0.0.1 # (required) server address.
+  port: 9000      # (required) server port. For native, defaults to 9440 secure and 9000 insecure. For HTTP, defaults to 8443 secure and 8123 insecure.
+
+  protocol: native # (required) the protocol used for the connection. Can be set to "native" or "http".
+  secure: false    # set to true if the connection is secure.
+
+  username: default # the username used for authentication.
+
+  tlsSkipVerify:     <boolean> # skips TLS verification when set to true.
+  tlsAuth:           <boolean> # set to true to enable TLS client authentication.
+  tlsAuthWithCACert: <boolean> # set to true if CA certificate is provided. Required for verifying self-signed TLS certificates.
+
+secureJsonData:
+  password: secureExamplePassword # the password used for authentication.
+
+  tlsCACert:     <string> # TLS CA certificate
+  tlsClientCert: <string> # TLS client certificate
+  tlsClientKey:  <string> # TLS client key
+```
+
+Note that a `version` property is added when the configuration is saved from the UI. This shows the version of the plugin that the config was saved with.
+
+### HTTP Protocol
+
+More settings will be displayed if you choose to connect via the HTTP protocol.
```

Comment:
Should we explain the ability to attach custom headers, or is it a standard practice for all the plugins?